lib/plug/conn.ex

```diff
@@ -390,7 +401,7 @@ defmodule Plug.Conn do
   atoms is available in `Plug.Conn.Status`.
 
   Raises a `Plug.Conn.AlreadySentError` if the connection has already been
-  `:sent` or `:chunked`.
+  `:sent`, `:chunked` or `:upgraded`.
```

Comment:
A number of cases such as this (where the existing docs weren't totally precise about required states) have been fixed here as well.